### PR TITLE
Detect invalid saved state (deleted VM still running), log and rectify

### PR DESCRIPTION
**Pull request by gerboland**
I can't figure out how the user got it in that state, but at least we should avoid crashing.
Fixes bug #773

### CODE REVIEW

**Comment by ricab** [comment]
Trying to put things in more practical light, what happens now on `recover`? or `mount`? We have no idea. We are now in our own *UB land*. Most likely we'll get leaks and eventually crashes, whose origin is harder to understand or overcome.

File: src/daemon/daemon.cpp [hunk 1]
```diff
@@ -595,9 +595,9 @@ mp::Daemon::Daemon(std::unique_ptr<const DaemonConfig> the_config)
             config->vault->remove(name);
         }
 
-        if (spec.state == VirtualMachine::State::running && vm_instances[name]->state != VirtualMachine::State::running)
+        if (spec.state == VirtualMachine::State::running && !spec.deleted &&
```
Comment (ricab) [inline src/daemon/daemon.cpp]: I thought this a bit more over at lunch. If we're getting inconsistent state from disk, I think we should log and correct, (to stopped or unknown).

```c++
if(spec.state == VirtualMachine::State::running)
    if(spec.deleted)
    {
        // log
        spec.state = VirtualMachine::State::unknown;
    }
    else if(vm_instances[name]->state != VirtualMachine::State::running)
    {
        [...]
    }

File: src/daemon/daemon.cpp [hunk 2]
```diff
@@ -595,6 +595,13 @@ mp::Daemon::Daemon(std::unique_ptr<const DaemonConfig> the_config)
             config->vault->remove(name);
         }
 
+        // FIXME: somehow we're writing contradictory state to disk.
+        if (spec.deleted && spec.state != VirtualMachine::State::stopped)
+        {
+            mpl::log(mpl::Level::debug, category, fmt::format("{} is deleted but has state", name, spec.state));
+            spec.state = VirtualMachine::State::stopped;
```
Comment (townsend2010) [inline src/daemon/daemon.cpp]: Can't assign here as spec is a `const`.

**Comment by gerboland** [comment]
I'm gonna remove this as I think I've a hand on the core problem

**Pull request closed without merging**